tests/jni_global_refs.rs

```diff
@@ -0,0 +1,129 @@
+#![cfg(feature = "invocation")]
+extern crate error_chain;
+extern crate jni;
+
+use std::sync::{Arc, Barrier, Once, ONCE_INIT};
+use std::thread::spawn;
+
+use error_chain::ChainedError;
+use jni::{InitArgsBuilder, JNIEnv, JNIVersion, JavaVM};
+use jni::errors::Result;
+use jni::objects::AutoLocal;
+use jni::objects::JValue;
+
+
+pub fn jvm() -> &'static Arc<JavaVM> {
+    static mut JVM: Option<Arc<JavaVM>> = None;
+    static INIT: Once = ONCE_INIT;
+
+
+    INIT.call_once(|| {
+        let jvm_args = InitArgsBuilder::new()
+            .version(JNIVersion::V8)
+            .option("-Xcheck:jni")
+            .option("-Xdebug")
+            .build()
+            .unwrap_or_else(|e| {
+                panic!(format!("{}", e.display_chain().to_string()));
+            });
+
+        let jvm = JavaVM::new(jvm_args).unwrap_or_else(|e| {
+            panic!(format!("{}", e.display_chain().to_string()));
+        });
+
+        unsafe {
+            JVM = Some(Arc::new(jvm));
+        }
+    });
+
+    unsafe { JVM.as_ref().unwrap() }
+}
+
+fn print_exception(env: &JNIEnv) {
+    let exception_occurred = env.exception_check()
+        .unwrap_or_else(|e| panic!(format!("{:?}", e)));
+    if exception_occurred {
+        env.exception_describe()
+            .unwrap_or_else(|e| panic!(format!("{:?}", e)));
+    }
+}
+
+fn unwrap<T>(env: &JNIEnv, res: Result<T>) -> T {
+    res.unwrap_or_else(|e| {
+        print_exception(&env);
+        panic!(format!("{}", e.display_chain().to_string()));
+    })
+}
+
+#[test]
+pub fn global_ref_works_in_other_threads() {
```

Comment:
I'd simplify the test to make it easier to understand by reviewers & future maintainers:

1. Use `incrementAndGet` in both threads.
2. Extract a magic `10000` in a constant (e.g., `incrementsInEachThread`, or `numIncrements`).
3. Use the same constant in both loops (no `10000 + 1`, because that is likely to confuse readers).
4. assert that the resulting value is equal to the number of threads (2) multiplied by the number of increments.

Reply:
2. Everything went to this, but I somehow forgot to finish it.